### PR TITLE
Sync's UX fixes

### PR DESCRIPTION
**Pull request by mdmohsin7**
https://github.com/user-attachments/assets/902475a6-4a0f-4ab0-9113-79d4cdfff095

https://github.com/user-attachments/assets/f5bd7b79-733b-4659-9292-9eaa79ff4c31

## Release Notes

- **New Features**
	- Enhanced user experience during memory syncing with updated status indicators, including a loading indicator and a message "Finalizing your memories."
	- Introduced a reprocessing option for discarded memories in the memory list.
	- Added a new function for converting seconds into a human-readable format.
	- Improved the UI for syncing memories with a more structured display of device type, ETA, and a sync button.

- **Bug Fixes**
	- Improved handling of memory fetching states, ensuring accurate loading indicators.

- **Refactor**
	- Streamlined memory list handling by removing unnecessary null checks.
	- Updated string handling methods for better data presentation.
	- Enhanced control flow for sync status determination and display.
	- Improved organization of code related to displaying and grouping memory items.

These changes aim to provide a smoother and more intuitive experience while managing memories in the app.

### CODE REVIEW

File: app/lib/pages/memories/sync_page.dart [hunk 1]
```diff
@@ -280,26 +280,27 @@ class _SyncPageState extends State<SyncPage> {
     }
 
     return PopScope(
-      canPop: !Provider.of<MemoryProvider>(context, listen: false).isSyncing,
+      canPop: true,
       onPopInvoked: (didPop) {
         var provider = Provider.of<MemoryProvider>(context, listen: false);
```
Comment (beastoin) [inline app/lib/pages/memories/sync_page.dart:230]: We allow pop, btw the sync process should be stop if users exited the sync page. We haven't had the background syncs yet 🫣

Comment (mdmohsin7) [reply app/lib/pages/memories/sync_page.dart:230]: The sync state is handled by provider and provider isn't disposed until the user closes the app. So the sync should keep running even if the user is on another screen. That is way changed the text as well (to only mention not to close the app)

Comment (mdmohsin7) [reply app/lib/pages/memories/sync_page.dart:230]: Lmk if I should revert it

Comment (beastoin) [reply app/lib/pages/memories/sync_page.dart:230]: would be great if we could indicate it somewhere that the sync is running, and when the provider closes, we could gratefully stop the sync(e.g notify sd card to stop sync) @mdmohsin7 any idea?

Comment (mdmohsin7) [reply app/lib/pages/memories/sync_page.dart:230]: We can definitely have some sync in progress indicator maybe through a notification without cluttering the existing UI. What do you think? @beastoin 

Provider is only disposed when the app is restarted or the app is closed completely

**Comment by beastoin** [comment]
lgtm 🥳

**Pull request merged**